### ISSUE TITLE
UserState throws an unexpected NullReference exception

### ISSUE THREAD

**Issue by kranid** [microsoft/botbuilder-dotnet#2457]
## Version
SDK v4
## Describe the bug
UserState throws  nullReference exception If pass argument force =true to its method SaveChangesAsync even if UserState has not changed.
## To Reproduce

1. Create an instance of UserState class
2. Call method UserState.SaveChangeAsync(force:true) '....'
## Actual behaveor
UserState Throws NullReference exception without any specific messages 

## Expected behavior
throws exception with a message that provides some information about reasons of this exception. 
## Additional context
This problem is caused by the following code from the class [BotState.cs](https://github.com/microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Builder/BotState.cs):
```
            var cachedState = turnContext.TurnState.Get<CachedBotState>(_contextServiceKey);
            if (force || (cachedState != null && cachedState.IsChanged()))
            {
                var key = GetStorageKey(turnContext);
```

**Comment by daveta** [comment]
Hi @kranid, thanks for filing this issue!  We'll investigate fixing this in the 4.6 timeframe.  In the meantime, feel free to submit a PR for this!

**Comment by EricDahlvang** [comment]
Thank you for reporting this @kranid 

This fix has been merged, and will go out with 4.6:
https://github.com/microsoft/botbuilder-dotnet/pull/2484
